numpy/core/tests/test_multiarray.py

```diff
@@ -9302,6 +9302,9 @@ def test_start_stop_kwarg(self):
         assert len(keyword_start_stop) == 6
         assert_array_equal(keyword_stop, keyword_zerotostop)
 
+    def test_rejects_string(self):
+        assert_raises(TypeError, np.arange, step=2, dtype=str)
+
```

Comment:
This test isn't quite right: I think you want the `stop` kwarg, not `step` (this already raises a `TypeError` on main - calling `arange` without `stop` raises a TypeError).

Also, since this is a new test I'd advocate for using `pytest.raises` instead of `assert_raises`, which is (IMO) more readable and would've caught the problem with this test if we used the `match` argument to verify the exception message:

```suggestion
    def test_rejects_string(self):
        with pytest.raises(TypeError, match="string types for arange"):
            np.arange(2, dtype=str)
```